### PR TITLE
[Rpc]Add processOptions in worker.config

### LINKED ISSUE #6671: Add support in worker.config to specify number of worker process count or opt-in to set worker count to number of core available on the VM
This came up while investigating some CRIs. The current default is one no matter how many cores are available on the VM.
And customers should be aware of the FUNCTIONS_WORKER_PROCESS_COUNT app setting to be able to increase their number of language workers for their specific scenarios.

For better throughput, the default should be set to the number of cores available. For Consumption  this will still be one but this will be different for Premium and App Service plans.

### PR DESCRIPTION
**Pull request by pragnagopa**
### Issue describing the changes in this PR

resolves #6671 

### Pull request checklist

* [x] My changes **do not** require documentation changes
    * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
    * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
    * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

Added support for customize multiple worker processes in `worker.config.json`

```json
{
  "description": {
    "language": "node",
    "extensions": [ ".js" ],
    "defaultExecutablePath": "node",
    "defaultWorkerPath": "dist/src/nodejsWorker.js"
  },
  "processOptions": {
    "processCount": 5,
    "maxProcessCount": 12,
    "processStartUpInterval": "00:00:05"
  }
}
```

If `processOptions` section is not specified,  following defaults will be applied:

`processCount` :  1
`maxProcessCount` : 10
`processStartUpInterval` : 00:00:10

### CODE REVIEW

**Comment by pragnagopa** [comment]
if any of you have suggestions for better section name for `processOptions` - please let me know.

File: src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs [hunk 1]
```diff
@@ -50,6 +48,7 @@ internal class RpcFunctionInvocationDispatcher : IFunctionInvocationDispatcher
         private IEnumerable<FunctionMetadata> _functions;
         private ConcurrentStack<WorkerErrorEvent> _languageWorkerErrors = new ConcurrentStack<WorkerErrorEvent>();
         private CancellationTokenSource _processStartCancellationToken = new CancellationTokenSource();
+        private int _debouceMilliSeconds = (int)TimeSpan.FromSeconds(10).TotalMilliseconds;
```
Comment (AnatoliB) [inline src/WebJobs.Script/Workers/Rpc/FunctionRegistration/RpcFunctionInvocationDispatcher.cs]: Typo: "debouNce"

**Comment by AnatoliB** [comment]
I have a concern that does not directly block this specific PR, but let's keep this in mind before workers actually start opting in. We previously talked about adding more safeguards before spawning more processes by default, such as checking memory consumption. I think we should introduce this check _before_ workers start opting in. Without this check, opting in cannot be a per-worker or per-language decision: if an app consumes a lot of memory, this have little to do with the language choice in many cases. Are we going to have this check soon?

**Comment by pragnagopa** [comment]
> checking memory consumption. I think we should introduce this check before workers start opting in. Without this check, opting in cannot be a per-worker or per-language decision: if an app consumes a lot of memory, this have little to do with the language choice in many cases. Are we going to have this check soon?

Tracking issue: https://github.com/Azure/azure-functions-host/issues/6704 . But I do not have an ETA yet.

Yes, I agree it makes sense to wait for the check to be added before workers can opt-in. I believe this PR would allow @vrdmr to run perf tests customizing number of workers and debouce interval. I would like get feedback from @vrdmr before prioritizing issue #6704

**Review by AnatoliB — APPROVED** [review]

**Comment by vrdmr** [comment]
> Yes, I agree it makes sense to wait for the check to be added before workers can opt-in. I believe this PR would allow @vrdmr to run perf tests customizing the number of workers and debounce intervals. I would like to get feedback from @vrdmr before prioritizing issue #6704

Thanks, @pragnagopa for the changes. This definitely helps me to do tests. The current configuration still requires hardcoding the `processCount` and we cannot choose based on the # cores on the node. Is there a way would you recommend to generate the process counts based on the number of cores?

**Comment by pragnagopa** [comment]
Missed adding this as part of PR description. `processOptions` exposes `SetWorkerCountToNumberOfCpuCores`  property. Setting this `true` will set the number of workers to cpu cores.

Here a sample worker.config.json
```json
{
  "description": {
    "language": "node",
    "extensions": [ ".js" ],
    "defaultExecutablePath": "node",
    "defaultWorkerPath": "dist/src/nodejsWorker.js"
  },
  "processOptions": {
    "setWorkerCountToNumberOfCpuCores": true,
    "processStartUpInterval": "00:00:05"
  }
}
```

**Comment by vrdmr** [comment]
Awesome! 👍 

Thanks a lot, @pragnagopa for the change. Cannot wait to try this out now.

**Pull request merged**